sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-obo/README.md

```diff
@@ -48,68 +55,44 @@ We need to ensure that this [environment checklist][ready-to-run-checklist] is c
+### Configure your middle-tier Web API A
+1. You can follow *Web API A* sample to add `ResourceAccessGraph.Read`, `ResourceAccessGraphCustomResources.Read` scopes. Convention current application id url is `api://xxxxx-xxxx-xxxxxx`.(**NOTE**: In the **Web API A** `ResourceAccessGraph.Read`, `ResourceAccessGraphCustomResources.Read` scopes should be added to the **Web APP**.)![API Permissions](docs/image-resource-server-obo-add-scope.png)
+2. Select **API permissions** > **Add a permission** > **My APIs**, select *Web API B* application name. ![Select MyAPIs](docs/image-select-myapis.png)
+3. **Delegated permissions** is selected by default， Select **File** > **File.Read** permission, select **Add permission** to complete the process.![Add Permissions](docs/image-add-permissions.png)
+4. Add **Web API B** permissions to **Web API A** and grant admin consent.![API Permissions](docs/image-add-grant-admin-consent.png)
 
-#### Add Custom API Permission
-
-1. Select **Azure Active Directory** > **App registrations**, and then select your current sample application (not your web API).
-
-2. Select **API permissions** > **Add a permission** > **My APIs**, select *azure-spring-boot-sample-active-directory-resource-server* application name.
-
-3. **Delegated permissions** is selected by default， Select **File** > **File.Read** permission, select **Add permission** to complete the process.
-
-### Grant consent for your tenant
-Respectively grant admin consent to the Graph and Custom permissions. After adding as shown below:
-   ![API Permissions](resource/resource-server-obo-add-permissions.png)
+See [OAuth 2.0 On-Behalf-Of flow] for more information about OBO.
 
 ## Examples
-
-### Configure application.yaml
-
+### Configure application.yml
 ```yaml
 azure:
   activedirectory:
     resource-server:
       obo:
         enabled: true
-    client-id: [resource-server-application-client-id]
-    client-secret: [resource-server-application-client-secret]
-    tenant-id: [tenant-id-registered-by-application]
-    app-id-uri: api://sample-client-id
+    client-id: <Web-API-A-client-id>
+    client-secret: <Web-API-A-client-secret>
+    tenant-id: <tenant-id-registered-by-application>
+    app-id-uri: <Web-API-A-app-id-url>
     authorization:
       graph:
         scopes:
           - https://graph.microsoft.com/User.Read
       custom:
         scopes:
-          - api://custom-client-id/File.read
+          - <Web-API-B-app-id-url>/File.Read
 ```
 
-### Run with Maven 
-
+### Run with Maven
 ```shell
 cd azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-obo
 mvn spring-boot:run
 ```
 
-### Access the Web API
-First, you need to obtain an access token to access Sample API.
-- API will Only call Graph resource. 
+### Access the Web API A
+First, you need to obtain an access token to access Sample Web API A.
```

Comment:
Maybe we could remove the `Sample` here.

Reply:
You are right.